modules/xfem/include/userobjects/PointValueAtXFEMInterface.h

```diff
@@ -0,0 +1,105 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef POINTVALUEATXFEMINTERFACE_H
+#define POINTVALUEATXFEMINTERFACE_H
+
+#include "GeneralUserObject.h"
+#include "ElementPairLocator.h"
+#include "MooseVariableFE.h"
+
+#include "libmesh/vector_value.h"
+
+// Forward Declarations
+class PointValueAtXFEMInterface;
+class XFEM;
+class LineSegmentCutSetUserObject;
+
+template <>
+InputParameters validParams<PointValueAtXFEMInterface>();
+
+class PointValueAtXFEMInterface : public GeneralUserObject
+{
+public:
+  PointValueAtXFEMInterface(const InputParameters & parameters);
+
+  virtual ~PointValueAtXFEMInterface() {}
+
+  virtual void initialize();
+  virtual void execute();
+  virtual void finalize();
+
+  std::map<unsigned int, Real> getValueAtPositiveLevelSet() const
+  {
+    return _values_positive_level_set_side;
+  };
+
+  std::map<unsigned int, Real> getValueAtNegativeLevelSet() const
+  {
+    return _values_negative_level_set_side;
+  };
+
+  std::map<unsigned int, RealVectorValue> getGradientAtPositiveLevelSet() const
+  {
+    return _grad_values_positive_level_set_side;
+  };
+
+  std::map<unsigned int, RealVectorValue> getGradientAtNegativeLevelSet() const
+  {
+    return _grad_values_negative_level_set_side;
+  };
+
+  unsigned int numberPoints() const { return _points.size(); };
+
+protected:
+  /**
+   * Find the element in the element pairs that contains the point in its physical domain.
+   * @param p The point in physical space
+   * @param positive_level_set True if the physical domain is in positive level set region
+   * @return The Elem containing the point or NULL if this processor doesn't contain an element that
+   * contains this point.
+   */
+  const Elem * getElemContainingPoint(const Point & p, bool positive_level_set);
+
+  /// The Mesh we're using
+  MooseMesh & _mesh;
+
+  /// The points to evaluate at
+  std::vector<Point> _points;
+
```

Comment:
Add one-line doxygen comment to this and a few variables below

Reply:
Done